File: scripts/crons/cron1.js
var	Suscriptions = require('payparrot_models/objects/suscriptions'),
	_ = require('underscore'),
	async = require('async'),
	queue = require('payparrot_models/libs/queue');

var first_payment = function(suscription, callback){
	queue.createMessage('payments', {
			MessageBody: JSON.stringify({'suscription_id': suscription._id, 'account_id': suscription.account_id, 'parrot_id': suscription.parrot_id})
		}, function(err){
			if(!err){
				Suscriptions.update(
					{'_id': suscription._id}, {'$set': {'first_tweet': true}},
					function(){
						callback();
					}
				);
			}else{
				callback();
			}
		}
	);
};

var db = require('payparrot_models/libs/mongodb').connect({});
Suscriptions.find({'active': true, 'first_tweet': false}, {}, function (err, suscriptions){
	async.forEach(suscriptions, function(item, callback){first_payment(item,callback)}, function(err){
		db.connection.close();
	});
});

File: scripts/payparrot_scripts/crons/cron1.py
# -*- coding: utf-8 -*-

from payparrot_dal import Subscriptions
from payparrot_dal.mongodb import connect
from payparrot_dal.queue import Queue
from payparrot_scripts.crons.utils import log 

def main():
    connection = None
    try:
        log('cron1', 'Starting')
        connection, db = connect()
        subscriptions = Subscriptions.find(db, {'active': True, 'first_tweet': False})
        for subscription_raw in subscriptions:
            created_message = Queue.insert(
                'payments', 
                {
                    'subscription_id': str(subscription_raw['_id']),
                    'account_id': str(subscription_raw['account_id']),
                    'parrot_id': str(subscription_raw['parrot_id'])
                }
            )
            log('cron1', 'Payment queued %s' % created_message.id, subscription_raw['_id'])
            db.subscriptions.update({'_id': subscription_raw['_id']}, {'$set': {'first_tweet': True}})
            log('cron1', 'Subscription updated', subscription_raw['_id'])
    finally:
        if connection:
            connection.close()


if __name__ == '__main__':
    main()